de.fhdo.lemma.service.open_api/src/de/fhdo/lemma/service/open_api/LemmaDataSubGenerator.xtend

```diff
@@ -0,0 +1,197 @@
+            )
+            return newListStructure
+        } else {
+            return foundObject
+        }
+    }
+
+    def void addDataFieldsToDataStructure(
+        DataStructure ds, String structureName, Schema structureSchema
+    ) {
+        if (structureSchema.properties !== null) {
+            structureSchema.properties.forEach[
+                name, schema|addDataFieldsToDataStructure(ds, name, schema)
+            ]
+        } else {
+            val newDataField = generateDataField(structureName, structureSchema)
+            ds.dataFields.add(newDataField)
+        }
+    }
+
+    def addDataFieldsToListStructure(
+        ListType ds, String structureName, Schema structureSchema
+    ) {
+        val newDataField = generateDataField(structureName, structureSchema)
+        ds.dataFields.add(newDataField)
+    }
+
+    def DataField generateDataField(
+        String structureName, Schema structureSchema
+    ) {
+        val newDataField = DATA_FACTORY.createDataField
+        newDataField.name = StringUtils.lowerCase(structureName)
+
+        switch (structureSchema.type) {
+            case "integer": {
+                newDataField.primitiveType = OpenApiUtil.deriveIntType(structureSchema.format)
+            }
+            case "number": {
+                newDataField.primitiveType = OpenApiUtil.deriveNumberType(structureSchema.format)
+            }
+            case "string": {
+                newDataField.primitiveType = OpenApiUtil.deriveStringType(structureSchema.format)
+            }
+            case "boolean": {
+                newDataField.primitiveType = DATA_FACTORY.createPrimitiveBoolean
+            }
+            case "array": {
+                val itemSchema = structureSchema as ArraySchema
+                val listStructure = getOrCreateListStructure(
+                    myContext, structureName, itemSchema.items
+                )
+                newDataField.complexType = listStructure
+            }
+            case null: {
+                // If Attribute is actually a reference...
+                if (structureSchema.get$ref !== null) {
+                    val s = structureSchema.get$ref
+                    val key = s.substring(s.lastIndexOf("/") + 1)
+                    val value = openAPI.components.schemas.get(key)
+                    if (key !== null && value !== null) {
+                        newDataField.complexType = getOrCreateDataStructure(
+                            myContext, key, value
+                        )
+                    }
+                }
+            }
+            default:
+                throw new IllegalArgumentException(
+                    "schema type " + structureSchema.type + " not supported"
+                )
+        }
+        return newDataField
+    }
+
+    def DataStructure findDataStructure(Context context, String name) {
+        val searchName = '''«context.buildQualifiedName(seperator)».«name»'''
```

Comment:
Don't use template syntax here. Furthermore, this statement makes the `seperator` constant even more redundant because you hard-code the second dot (which is also the value of the constant).